### PR TITLE
[SYCLomatic] Support migration of cub::{SHL_ADD, SHR_ADD, BFE, BFI}

### PR DESCRIPTION
**Pull request by yihwang-nv**
Depends on #1525
Depends on #1481

### CODE REVIEW

**Review by tangjj11 — APPROVED** [review]

File: clang/test/dpct/cub/intrinsic/shl_add.cu [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// UNSUPPORTED: cuda-8.0, cuda-9.0, cuda-9.1, cuda-9.2, cuda-10.0, cuda-10.1, cuda-10.2
+// UNSUPPORTED: v8.0, v9.0, v9.1, v9.2, v10.0, v10.1, v10.2
+// RUN: dpct -in-root %S -out-root %T/intrinsic/shl_add %S/shl_add.cu --cuda-include-path="%cuda-path/include" -- -std=c++14 -x cuda --cuda-host-only
+// RUN: FileCheck --input-file %T/intrinsic/shl_add/shl_add.dp.cpp --match-full-lines %s
+// RUN: %if build_lit %{icpx -c -fsycl %T/intrinsic/shl_add/shl_add.dp.cpp -o %T/intrinsic/shl_add/shl_add.dp.o %}
+
+// CHECK:#include <sycl/sycl.hpp>
+// CHECK:#include <dpct/dpct.hpp>
```
Comment (tangjj11) [inline clang/test/dpct/cub/intrinsic/shl_add.cu:8]: maybe need check no cub header.

Comment (yihwang-nv) [reply clang/test/dpct/cub/intrinsic/shl_add.cu:8]: It's covered by other test case

**Review by zhimingwang36 — APPROVED** [review]
LGTM

File: clang/lib/DPCT/APINames_CUB.inc [hunk 2]
```diff
@@ -200,10 +200,10 @@ ENTRY(cub::LoadDirectWarpStriped, cub::LoadDirectWarpStriped, false, NO_FLAG, P4
 ENTRY(cub::StoreDirectWarpStriped, cub::StoreDirectWarpStriped, false, NO_FLAG, P4, "Comment")
 
 // PTX intrinsics
-ENTRY(cub::SHR_ADD, cub::SHR_ADD, false, NO_FLAG, P4, "Comment")
-ENTRY(cub::SHL_ADD, cub::SHL_ADD, false, NO_FLAG, P4, "Comment")
-ENTRY(cub::BFE, cub::BFE, false, NO_FLAG, P4, "Comment")
-ENTRY(cub::BFI, cub::BFI, false, NO_FLAG, P4, "Comment")
+ENTRY(cub::SHR_ADD, cub::SHR_ADD, true, NO_FLAG, P4, "Successful")
+ENTRY(cub::SHL_ADD, cub::SHL_ADD, true, NO_FLAG, P4, "Successful")
+ENTRY(cub::BFE, cub::BFE, true, NO_FLAG, P4, "Successful")
```
Comment (zhimingwang36) [inline clang/lib/DPCT/APINames_CUB.inc:205]: Please add lit test for BFE and BFI.

Comment (yihwang-nv) [reply clang/lib/DPCT/APINames_CUB.inc:205]: done

**Pull request merged**